### PR TITLE
Add UVI API

### PR DESCRIPTION
**Pull request by peter279k**
# Change log
- Add UVI API.
- Remove allow_failures configuration.

I'm not sure it's the proper way to implement the UVI API.
Please review this.

Thanks.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,11 +14,6 @@ php:
 
 matrix:
   fast_finish: true
-  allow_failures:
-    - php: nightly
-
-before_install:
-  - if [[ ! $TRAVIS_PHP_VERSION = hhvm* ]]; then phpenv config-rm xdebug.ini || echo "xdebug not available"; fi
```
Comment (cmfcmf) [inline .travis.yml]: Could you please revert these changes or reason why you made them?

**Comment by peter279k** [comment]
Hi @cmfcmf, I remove the nightly because I think the master branch should be passed by the nightly,
And it shouldn't allow failures.

Secondly, I move the if [[ ! $TRAVIS_PHP_VERSION = hhvm* ]]; then phpenv config-rm xdebug.ini || echo "xdebug not available"; fi to the install step.

Thanks.

**Comment by cmfcmf** [comment]
> I think the master branch should be passed by the nightly,

I must disagree here, I've reverted this change in my PR #99.
Please take a look at #99. It includes most of your changes plus a few adjustments. Thank you for your work!
